examples/tic-tac-toe/cli/src/client.rs

```diff
@@ -0,0 +1,749 @@
+        let tx = self.wallet.sign_transaction(&data);
+        let SuiTransactionBlockResponse {
+            object_changes: Some(object_changes),
+            ..
+        } = self.execute_transaction(tx).await?
+        else {
+            bail!("Can't find Game ID");
+        };
+
+        let Some(game_id) = object_changes.into_iter().find_map(|change| {
+            let ObjectChange::Created {
+                object_type,
+                object_id,
+                ..
+            } = change
+            else {
+                return None;
+            };
+
+            if ObjectID::from(object_type.address) != self.package {
+                return None;
+            }
+
+            if object_type.name.as_str() != "Game" {
+                return None;
+            }
+
+            Some(object_id)
+        }) else {
+            bail!("Can't find Game ID");
+        };
+
+        Ok(game_id)
+    }
+
+    /// Like `build_tx_data_with_sponsor`, but without a sponsor.
+    async fn build_tx_data(
+        &self,
+        sender: SuiAddress,
+        tx: ProgrammableTransaction,
+    ) -> Result<TransactionData> {
+        self.build_tx_data_with_sponsor(sender, None, tx).await
+    }
+
+    /// Do gas estimation and coin selection to create a `TransactionData` from a
+    /// `ProgrammableTransaction`. If `sponsor` is provided, it will be used as the gas sponsor, and
+    /// coin selection will fetch coins owned by this address, otherwise coins will be selected from
+    /// the `sender`'s owned objects.
+    async fn build_tx_data_with_sponsor(
+        &self,
+        sender: SuiAddress,
+        sponsor: Option<SuiAddress>,
+        tx: ProgrammableTransaction,
+    ) -> Result<TransactionData> {
+        let client = self.client().await?;
+
+        let max_budget = self.max_gas_budget().await?;
+
+        let gas_price = self
+            .wallet
+            .get_reference_gas_price()
+            .await
+            .context("Error fetching reference gas price")?;
+
+        let tx_kind = TransactionKind::ProgrammableTransaction(tx);
+
+        // Gas Estimation
+        let tx_data = client
+            .transaction_builder()
+            .tx_data_for_dry_run(
+                sender,
+                tx_kind.clone(),
+                max_budget,
+                gas_price,
+                /* gas_payment */ None,
+                /* gas_sponsor */ None,
+            )
+            .await;
+
+        let DryRunTransactionBlockResponse { effects, .. } = client
+            .read_api()
+            .dry_run_transaction_block(tx_data.clone())
+            .await
+            .context("Error estimating gas budget")?;
+
+        let gas_used = effects.gas_cost_summary();
+        let overhead = 1000 * gas_price;
+        let net_used = gas_used.net_gas_usage();
+        let computation = gas_used.computation_cost;
+
+        let budget = overhead + (net_used.max(0) as u64).max(computation);
```

Comment:
this tells me that the overhead function in the CLI `dry_run` logic should be part of the SDK.